### PR TITLE
Add LevelSetEdges in ConstraintGraphFactory

### PR DESCRIPTION
**Pull request by florent-lamiraux**
- make C++ and python construction of constraint graphs closer.
requires https://github.com/humanoid-path-planner/hpp-manipulation/pull/80 and https://github.com/humanoid-path-planner/hpp-corbaserver/pull/120.

### CODE REVIEW

File: idl/hpp/manipulation_idl/_graph.idl [hunk 1]
```diff
@@ -107,16 +106,47 @@ module hpp {
...
+        //* bool res(getT()->generateTargetConfig(_qStart, _q));
+        //* q = *vectorToFloatSeq(_q);
+        //* return res;
+
         core_idl::SteeringMethod getSteeringMethod () raises (Error);
         //-> steeringMethod
         core_idl::PathValidation getPathValidation () raises (Error);
         //-> pathValidation
 
+        // deprecated
```
Comment (jmirabel) [inline idl/hpp/manipulation_idl/_graph.idl:124]:
```suggestion
        /// \deprecated renamed to \ref targetConstraint
```

File: idl/hpp/manipulation_idl/device.idl [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      //* return hpp::corbaServer::toNames_t (__return__);
+
+      Handle getHandle (in string name) raises(Error);
+      //* std::string _name (name);
+      //* hpp::manipulation::HandlePtr_t __return__
+      //* (getT()->handles.get (_name));
+      //* return makeServantDownCast
+      //* <hpp::corbaServer::manipulation_impl::Handle,
+      //* hpp::corbaServer::manipulation_impl::Handle>
+      //* (server_, __return__)._retn();
```
Comment (jmirabel) [inline idl/hpp/manipulation_idl/device.idl:46]:
```suggestion
      //-> handles.get
```

File: idl/hpp/manipulation_idl/_graph.idl [hunk 3]
```diff
@@ -107,16 +106,47 @@ module hpp {
...
+                                           in floatSeq qLeaf,
+                                           inout floatSeq q) raises(Error);
+        //* using hpp::corbaServer::floatSeqToVector;
+        //* using hpp::corbaServer::vectorToFloatSeq;
+        //* vector_t _qStart(floatSeqToVector(qStart));
+        //* vector_t _qLeaf(floatSeqToVector(qLeaf));
+        //* vector_t _q(floatSeqToVector(q));
+        //* bool res(getT()->generateTargetConfigOnLeaf(_qStart, _qLeaf, _q));
+        //* q = *vectorToFloatSeq(_q);
+        //* return res;
```
Comment (jmirabel) [inline idl/hpp/manipulation_idl/_graph.idl:147]:
```suggestion
```

File: idl/hpp/manipulation_idl/_graph.idl [hunk 4]
```diff
@@ -107,16 +106,47 @@ module hpp {
...
 
+        boolean generateTargetConfig(in floatSeq qStart, inout floatSeq q)
+          raises(Error);
+        //* using hpp::corbaServer::floatSeqToVector;
+        //* using hpp::corbaServer::vectorToFloatSeq;
+        //* vector_t _qStart(floatSeqToVector(qStart));
+        //* vector_t _q(floatSeqToVector(q));
+        //* bool res(getT()->generateTargetConfig(_qStart, _q));
+        //* q = *vectorToFloatSeq(_q);
+        //* return res;
```
Comment (jmirabel) [inline idl/hpp/manipulation_idl/_graph.idl:117]:
```suggestion
```

File: idl/hpp/manipulation_idl/device.idl [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      //* std::vector<bool> _mask(hpp::corbaServer::boolSeqToVector(mask, 6));
+      //* getT()->mask(_mask);
+    }; // interface Handle
+    interface Device : pinocchio_idl::Device
+    {
+      Names_t getHandleNames() raises(Error);
+      //* typedef core::Container <manipulation::HandlePtr_t>::key_type K_t;
+      //* typedef std::vector<K_t> strings;
+      //* strings __return__ (getT()->handles.template getKeys <strings>());
+      //* return hpp::corbaServer::toNames_t (__return__);
```
Comment (jmirabel) [inline idl/hpp/manipulation_idl/device.idl:37]:
```suggestion
      //-> handles.template getKeys < std::vector<std::string> >
```

**Comment by florent-lamiraux** [comment]
I have taken your suggestions and tested. Everything is all right.

**Pull request merged**